### PR TITLE
Editor: Wire PHP tests for block editor settings

### PR DESCRIPTION
**Pull request by gziolo**
## Description

Follow-up for #30245.

I missed one tiny but extremely important detail, PHPUnit test files need to end with `-test.php`. The fact that I extracted code from https://github.com/WordPress/wordpress-develop/pull/1118 didn't help as well 😅 

I had to update some tests in comparison to the version in WordPress core to account for the hooks that are present in the Gutenberg plugin.

## How has this been tested?

`npm run test-php` - ensure that tests run and pass.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: phpunit/block-editor-test.php [hunk 1]
```diff
@@ -243,7 +244,7 @@ function test_get_default_block_editor_settings() {
 	 */
 	function test_get_block_editor_settings_returns_default_settings() {
 		$this->assertSameSets(
-			gutenberg_get_block_editor_settings( 'post-editor' ),
+			gutenberg_get_block_editor_settings( 'my-editor' ),
```
Comment (gziolo) [inline phpunit/block-editor-test.php:247]: When using `post-editor` all hooks from the Gutenberg plugin get applied as well.

Comment (ntsekouras) [reply phpunit/block-editor-test.php:247]: Why this change is need though?

**Review by ntsekouras — APPROVED** [review]
Nice 👍

File: phpunit/block-editor-test.php (hunk 1 above)
Comment (gziolo) [reply phpunit/block-editor-test.php:247]: At the moment the wrapper function returns the same structure as the default block editor settings. `post-editor` has special handling so it applies all existing filters present in the Gutenberg plugin.

Comment (ntsekouras) [reply phpunit/block-editor-test.php:247]: So it's just a change to not apply filters that we don't need for these tests or the filters interfere with the output? I'm not familiar with this :)

Comment (gziolo) [reply phpunit/block-editor-test.php:247]: Yes, those filters introduce noise for tests so I found a way to ignore them 😄 Eventually those tests will be run only in WP core and it won't be an issue anymore.

Comment (ntsekouras) [reply phpunit/block-editor-test.php:247]: Thanks for explaining Grzegorz!

**Pull request merged**